examples/basic/providers.tf

```diff
@@ -10,16 +10,10 @@ data "aws_eks_cluster_auth" "this" {
   name = module.eks_cluster.eks_cluster_id
 }
 
-provider "kubernetes" {
```

Comment:
Dont we need kubernetes provider ? in the examples we are calling module with `argo_helm_enabled = false`. In that case we use kubernetes provider for installation. If we remove this provider from `examples/basic/providers.tf` then `module "argocd_self_managed_kubernetes"` will fail.

Reply:
It is not possible to use kubernetes_manifest without an existing cluster so this installation method will fail anyway. I've update docs to make this more obvious and put the provider back for the case when the cluster is up and running.